runtimes/common/src/migrations.rs

```diff
@@ -58,12 +71,39 @@ where
 	}
 
 	#[cfg(feature = "try-runtime")]
-	fn post_upgrade(_state: sp_std::vec::Vec<u8>) -> Result<(), &'static str> {
+	fn post_upgrade(_state: sp_std::vec::Vec<u8>) -> Result<(), TryRuntimeError> {
+		use frame_support::ensure;
+
 		log::info!("RemoveInsecureRandomnessPallet::post_upgrade() checks 🔍");
-		if frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b"") {
-			Err("Storage in pallet_insecure_randomness_collective_flip is not empty after migration.")
-		} else {
-			Ok(())
-		}
+
+		ensure!(
+			!frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b""),
+			TryRuntimeError::Other(
+				"Storage in pallet_insecure_randomness_collective_flip is not empty after migration.",
+			)
+		);
+		Ok(())
+	}
+}
+
+pub struct BumpStorageVersion<T>(PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for BumpStorageVersion<T>
+where
+	T: frame_system::Config,
+	T: pallet_tips::Config,
+	T: cumulus_pallet_parachain_system::Config,
+	T: pallet_membership::Config<Instance2>,
+	T: cumulus_pallet_xcmp_queue::Config,
+	T: cumulus_pallet_dmp_queue::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
+		StorageVersion::new(4).put::<pallet_tips::Pallet<T>>();
+		StorageVersion::new(2).put::<cumulus_pallet_parachain_system::Pallet<T>>();
+		StorageVersion::new(2).put::<cumulus_pallet_xcmp_queue::Pallet<T>>();
+		StorageVersion::new(4).put::<pallet_membership::Pallet<T, Instance2>>();
+		StorageVersion::new(1).put::<cumulus_pallet_dmp_queue::Pallet<T>>();
+
+		<T as frame_system::Config>::DbWeight::get().reads_writes(0, 5)
```

Comment:
You can use the `.writes()` function directly.

Reply:
Sure: [a483e12](https://github.com/KILTprotocol/kilt-node/pull/541/commits/a483e12b418affb2d51cf88116833aab191e8560)